src/nixpacks/environment.rs

```diff
@@ -16,19 +17,26 @@ impl Environment {
     pub fn from_envs(envs: Vec<&str>) -> Result<Environment> {
         let mut environment = Environment::default();
         for env in envs {
-            let v: Vec<&str> = env.split('=').collect();
-            if v.len() == 1 {
+            let matches = Regex::new(r"([^=]*)(?:=?)([^=]*)([\s\S]*)")
```

Comment:
I don't think this regex is correct. Environment variable names cannot have any character. We should limit it to alphanum characters

Reply:
I was mirroring behaviour that was already there, which was splitting which would account for any character. But I'll implement.